### ISSUE TITLE
Mermaid chart not working on 0.3.0 docker

### ISSUE THREAD

**Issue by cromelex** [rishikanthc/markopolis#7]
Version 0.3.0 docker

I just tried a mermaid chart and I get the following error

> ERROR:markopolis.scripts.consume:Error getting remote content for Japan-Itinerary: 404 Client Error: Not Found for url: https://my.domain/notes/Japan-Itinerary/raw
> INFO:markopolis.scripts.consume:File Japan-Itinerary.md marked for publishing (new or unable to fetch remote content).

I noticed on the url the /notes/
Not sure if it is supposed to be there, but my environment variable MARKOPOLIS_TITLE="Notes"

Code for the mermaid chart
``` 
```mermaid
flowchart LR

  

A[Osaka 7-8] --> B[Tokyo 9-11]
B -.Nagano .-> C[Matsumoto]
C -.Nagano & Toyama.-> D[Takayama 12] <--> D1(Hida no Sato village)
B -.Nagano & Toyama.-> D
C <-.bus.-> D
D --Toyama---> E <--> D2([Onsen 14]) --> F
E[Kanazawa 13] ---> F[Kyoto 15-18] <--> F2(Uji) <--> F1(Nara)
F <-.-> F4(Himeji)```
````

It works even here on Github (same sample just for the sake of the example)
```mermaid
flowchart LR

  

A[Osaka 7-8] --> B[Tokyo 9-11]
B -.Nagano .-> C[Matsumoto]
C -.Nagano & Toyama.-> D[Takayama 12] <--> D1(Hida no Sato village)
B -.Nagano & Toyama.-> D
C <-.bus.-> D
D --Toyama---> E <--> D2([Onsen 14]) --> F
E[Kanazawa 13] ---> F[Kyoto 15-18] <--> F2(Uji) <--> F1(Nara)
F <-.-> F4(Himeji)
```

**Comment by rishikanthc** [comment]
I just pushed a new image. Can you try again. It's working on the test deployment.
https://markopolis.app/folder/Markdown-Syntax

**Comment by rishikanthc** [comment]
The error you are seeing is normal. That's the script checking if the md file exists on the server before pushing. That error is only for logging purposes.

**Comment by cromelex** [comment]
> I just pushed a new image. Can you try again.

Getting an error pulling that.

docker pull ghcr.io/rishikanthc/markopolis:0.3.0
0.3.0: Pulling from rishikanthc/markopolis
no matching manifest for linux/amd64 in the manifest list entries

**Comment by rishikanthc** [comment]
Sorry messed up the docker image. Can you try now ?

**Comment by cromelex** [comment]
Sorry still doesn't seem to be working for me. I am just getting a blank. Looking at the page's source code on the browser (Firefox 129) it seems to be there, but not being rendered.

Btw keep up the good work, sure there's a few things that need fixing but you are getting a lot done very quickly.

**Comment by rishikanthc** [comment]
That's super weird. I just checked on Firefox 129 and it loads fine for me. Are you seeing any console errors on the browser ?
Attaching a screenshot. Can you disable cache and refresh ?

<img width="878" alt="Screenshot 2024-08-15 at 1 37 08 PM" src="https://github.com/user-attachments/assets/24746bb9-2e55-42df-bc80-c8306c1c09de">

**Comment by rishikanthc** [comment]
Okay I just checked with your specific mermaid chart and it doesn't load. I will debug this. Thank you so much for testing everything out. This is very useful for me. Thanks for all the help.

**Comment by rishikanthc** [comment]
Found the bug ! Fixed it. Pull 0.3.1 
<img width="845" alt="Screenshot 2024-08-15 at 2 14 41 PM" src="https://github.com/user-attachments/assets/2964d176-5582-4891-a53a-cdaf4cec062c">

**Comment by cromelex** [comment]
Confirmed working! 😁 I checked with another mermaid chart I had (a complex timeline) and it is also rendering perfectly now. 
Thanks

**Comment by cromelex** [comment]
Actually, I've just checked on my desktop and while the chart is being rendered, it is currently super small. I think it might be related to the width. 
If I set it to top down instead of left-right it renders nicely.

Charts like timeline don't offer a top down, so they get very squeezed and look tiny.

**Comment by rishikanthc** [comment]
Yes this is because of the layout/design. Can you give me a couple sample charts so I can play around with the design a bit and see what works ?

**Comment by cromelex** [comment]
> Yes this is because of the layout/design. Can you give me a couple sample charts so I can play around with the design a bit and see what works ?

Found a nice random sample online that should give you a large chart to test with:

```
```mermaid
timeline
    section .NET Framework
        2000 - 2005 
             : .NET Framework 1.0
             : .NET Framework 1.0 SP1
             : .NET Framework 1.0 SP2
             : .NET Framework 1.1
             : .NET Framework 1.0 SP3
             : .NET Framework 2.0
        2006 - 2009 
             : .NET Framework 3.0
             : .NET Framework 3.5 
             : .NET Framework 2.0 SP 1 
             : .NET Framework 3.0 SP 1 
             : .NET Framework 2.0 SP 2 
             : .NET Framework 3.0 SP 2 
             : .NET Framework 3.5 SP 1
        2010 - 2015 
             : .NET Framework 4.0
             : .NET Framework 4.5
             : .NET Framework 4.5.1
             : .NET Framework 4.5.2
             : .NET Framework 4.6
             : .NET Framework 4.6.1
    section .NET Core
        2016 - 2017 
             : .NET Core 1.0
             : .NET Core 1.1
             : .NET Framework 4.6.2
             : .NET Core 2.0
             : .NET Framework 4.7
             : .NET Framework 4.7.1
        2018 - 2019 
             : .NET Core 2.1
             : .NET Core 2.2
             : .NET Framework 4.7.2             
             : .NET Core 3.0
             : .NET Core 3.1
             : .NET Framework 4.8
    section Modern .NET
        2020 : .NET 5
        2021 : .NET 6
        2022 : .NET 7
             : .NET Framework 4.8.1

```
```

**Comment by rishikanthc** [comment]
Thanks. Will work on this

**Comment by rishikanthc** [comment]
I have updated the frontend now to make it wider. The above example mermaid chart renders fully and I think it's readable. Here's a demo of it: https://markopolis.app/Markdown%20Syntax
The fix is available in the latest release.
Let me know if this works. I'm also thinking of adding a toggle to allow user to switch between full width and narrow modes.

**Comment by cromelex** [comment]
It is a lot better (that particular graph on the demo site is perfect) but I have an even wider timeline that still looks smalish. Added a screenshot just to give you an idea of how it looks.
![image](https://github.com/user-attachments/assets/31306625-a091-452a-9eba-01a5ab06d0e0)


I've extended the sample above just to give you more to work with.

```
```mermaid
timeline
    section .NET Framework
        2000 - 2005 
             : .NET Framework 1.0
             : .NET Framework 1.0 SP1
             : .NET Framework 1.0 SP2
             : .NET Framework 1.1
             : .NET Framework 1.0 SP3
             : .NET Framework 2.0
        2006 - 2009 
             : .NET Framework 3.0
             : .NET Framework 3.5 
             : .NET Framework 2.0 SP 1 
             : .NET Framework 3.0 SP 1 
             : .NET Framework 2.0 SP 2 
             : .NET Framework 3.0 SP 2 
             : .NET Framework 3.5 SP 1
        2010 - 2015 
             : .NET Framework 4.0
             : .NET Framework 4.5
             : .NET Framework 4.5.1
             : .NET Framework 4.5.2
             : .NET Framework 4.6
             : .NET Framework 4.6.1
    section .NET Core
        2016 - 2017 
             : .NET Core 1.0
             : .NET Core 1.1
             : .NET Framework 4.6.2
             : .NET Core 2.0
             : .NET Framework 4.7
             : .NET Framework 4.7.1
        2018 - 2019 
             : .NET Core 2.1
             : .NET Core 2.2
             : .NET Framework 4.7.2             
             : .NET Core 3.0
             : .NET Core 3.1
             : .NET Framework 4.8
    section Modern .NET
        2020 : .NET 5
        2021 : .NET 6
        2022 : .NET 7
             : .NET Framework 4.8.1
    section Future 
        2024: random1
        2025: random2
        2026
        2027
        2028
        2029
        2030: random random
```
```

**Comment by rishikanthc** [comment]
Got it. Can you tell me what resolution your monitor or screen is ? I don't think I have optimized the UI for anything larger than 1080p.

**Comment by cromelex** [comment]
Sorry should have mentioned it. 3840x2160 
I think this will be the common upper limit these days for most people. 
I know most people aren't using such resolutions so it's no big deal.

**Comment by rishikanthc** [comment]
@cromelex wanted to get your thoughts on something.
I'm looking to optimize the UI for long form content that can have larger width. Since I want to get more real-estate I'm thinking of getting rid of the left menu panel and instead use a breadcrumb style navigation or maybe a toggle/trigger based option.

Any thoughts or preferences ? I still want the UI/UX to be nice and aesthetic. So wanted to bounce off some ideas.

**Comment by cromelex** [comment]
Happy to share my thoughts from a user perspective.

Personally, I quite like the left menu panel. I like them, their intuitive, and there's a reason loads of websites/apps use them. For people who use Obsidian it also replicates the layout so feels like it should be there?
I do get your point though. Just from a quick look, the left side menu is typically 20% thinner on most sites/apps then what your current one does on markopolis.app . Gmail, Reddit, Google Photos, Home Assistant (lovelace)  etc are just a few sites/apps  I just checked. Their navigation bar is almost the exact same size across all of them. I'd start there.

On my phone I like how it gets hidden as well, but I agree that when it's hidden, it kind of feels like you're missing a navigation. Sure, the hamburger button is there (and you should keep it, it's "standard"), but it doesn't have the same ease of use. A breadcrumb navigation could probably make more sense there?  Ie, when you are in a vertical layout.
If you could convert them in a straightforward manner, you could use your existing logic - when the screen is narrow (ie, phone), hide the navigation bar and add the breadcrumbs. 

Just as an example, not really a large width website, but https://www.home-assistant.io/docs/automation/yaml/ does this where when you narrow the window it hides the menu and you still have the breadcrumbs. I though the breadcrumb was hidden in "landscape" mode but it actually still there, it is just not as noticeable visually on the bigger screen. 


So short version:
- 100% Keep the left menu panel but reduce it's width by 20% to start with.
-  Add breadcrumb navigation to provide better navigation on mobile.

**Comment by rishikanthc** [comment]
Have revamped the UI to handle large images and mermaid diagrams better. The overall layout should also scale nicely to larger resolutions. Let me know if this works.

**Comment by cromelex** [comment]
> Have revamped the UI to handle large images and mermaid diagrams better. The overall layout should also scale nicely to larger resolutions. Let me know if this works.

Cheers!
I'm away for a few days but once I'm back I'll update and give some feedback on how this is working on my 3840x2160 screen.
